lib/noticed/delivery_methods/fcm.rb

```diff
@@ -20,7 +20,7 @@ def deliver
         device_tokens.each do |device_token|
           post("#{BASE_URI}#{project_id}/messages:send", headers: {authorization: "Bearer #{access_token}"}, json: {message: format(device_token)})
         rescue ResponseUnsuccessful => exception
-          if exception.code == 404
+          if exception.response.parse.dig('error', 'code') == 404
```

Comment:
We should be able to use `exception.response.code` here instead of having to pull the code out of the JSON. I'm going to make an edit to the PR and see if that works.

Reply:
Hi @excid3 Thanks for your feedback, but I think google always return `200` as status code 
see this trace code returned from `fcm`
```ruby 
[2022-02-03T15:16:47.924301 #90775] DEBUG -- : [httplog] {"method":"POST","url":"http://www.googleapis.com:443/oauth2/v4/token","response_code":200,"benchmark":0.1543269995599985}
400 Bad Request
{
  "error": {
    "code": 400,
    "message": "The registration token is not a valid FCM registration token",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.firebase.fcm.v1.FcmError",
        "errorCode": "INVALID_ARGUMENT"
      }
    ]
  }
}
```

so I think we must check the `code` entry inside `error` object.

Reply:
That's the info I needed. :+1:

Can you update and include a comment in the code?

Reply:
The logs show `400 Bad Request` so it looks like the status code is `400`, not `200`?

I think we'll need to check if the code is `400` or `404` then.